### ISSUE TITLE
Files suddenly do get missing in coverage

### ISSUE THREAD

**Issue by on3iro** [istanbuljs/nyc#476]
Please use the template provided below, when reporting bugs:
-----

## Expected Behavior
All tested files should be included in the coverage report.

## Observed Behavior
I am currently writing a React/Redux application. While coverage in general seems to work fine,
sometimes files are suddenly ignored for no apparent reason. E.g. all my reducer and action files (all in separate `ducks.js` files) are no longer visible inside the report. 

Even a tested component has been excluded after changing and saving it. Although the only thing I did was to add another test. To be clear: **all tests are running as expected**. they are just no longer covered by coverage.

It seems to happen whenever I edit one of the original component source files.

I was able to explicitly include the `ducks.js` files, but could not manage to include the suddenly missing component as well. 

## Forensic Information

**Operating System:** t
I am currently on windows using the git-bash for all npm-related tasks and am editing inside the windows-subsystem with vim.

**Include rules in babelrc**:
```
{
  "presets": ["react", "es2015", "stage-0"],
  "env": {
    "test": {
      "plugins": [
        ["istanbul", {
          "include": [
            "**/src/*.js",
            "**/src/**/ducks.js"
          ],
          "exclude": [
            "**/*_test.js",
            "**/*browser.js"
          ]
        }]
      ]
    }
  }
}
```

**package.json**:
```
{
  "name": "",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "directories": {
    "doc": "doc"
  },
  "scripts": {
    "test": "cross-env NODE_ENV=test _mocha --recursive -R spec",
    "test_watch": "cross-env NODE_ENV=test mocha -w --recursive -R spec",
    "coverage": "nyc --all npm test && nyc report --reporter=html --reporter=text-summary",
    "start": "node server",
    "build": "rimraf dist && webpack --config ./webpack.production.config.js --progress --profile --colors",
    "eslint": "eslint ./"
  },
  "author": "",
  "repository": {
    "type": "git",
    "url": ""
  },
  "nyc": {
    "sourceMap": false,
    "instrument": false
  },
  "bugs": {
    "url": ""
  },
  "devDependencies": {
    "autoprefixer": "^6.5.3",
    "babel": "^6.5.2",
    "babel-cli": "^6.18.0",
    "babel-core": "^6.18.2",
    "babel-eslint": "^7.1.0",
    "babel-loader": "^6.2.7",
    "babel-plugin-istanbul": "^3.0.0",
    "babel-polyfill": "^6.16.0",
    "babel-preset-es2015": "^6.18.0",
    "babel-preset-react": "^6.16.0",
    "babel-preset-stage-0": "^6.16.0",
    "babel-register": "^6.18.0",
    "chai": "^3.5.0",
    "cross-env": "^3.1.3",
    "css-loader": "^0.25.0",
    "enzyme": "^2.6.0",
    "eslint": "^3.9.1",
    "eslint-loader": "^1.6.1",
    "eslint-plugin-react": "^6.6.0",
    "extract-text-webpack-plugin": "^1.0.1",
    "file-loader": "^0.9.0",
    "html-webpack-plugin": "^2.24.1",
    "istanbul": "^0.4.5",
    "jsdom": "^9.8.3",
    "jsdom-global": "^2.1.0",
    "json-loader": "^0.5.4",
    "jsx-loader": "^0.13.2",
    "mocha": "^3.2.0",
    "moxios": "^0.3.0",
    "node-sass": "^3.11.2",
    "nyc": "^10.0.0",
    "postcss-loader": "^1.1.1",
    "react-addons-test-utils": "^15.4.1",
    "redux-mock-store": "^1.2.1",
    "rimraf": "^2.5.4",
    "sass-loader": "^4.0.2",
    "sinon": "^1.17.6",
    "style-loader": "^0.13.1",
    "webpack": "^1.13.3",
    "webpack-dev-server": "^1.16.2"
  },
  "dependencies": {
    "axios": "^0.15.2",
    "babel-polyfill": "^6.16.0",
    "classnames": "^2.2.5",
    "dotenv": "^2.0.0",
    "immutable": "^3.8.1",
    "lodash": "^4.17.2",
    "moment": "^2.17.0",
    "react": "^15.4.1",
    "react-dom": "^15.4.1",
    "react-redux": "^4.4.6",
    "react-router": "^3.0.0",
    "redux": "^3.6.0",
    "redux-promise": "^0.5.3",
    "redux-thunk": "^2.1.0",
    "reselect": "^2.5.4"
  }
}

```

**Environment Information:** 
https://gist.github.com/on3iro/1db7d6b3f54bcf68a088d8e29ebb22cc

**Comment by bcoe** [comment]
@on3iro I'm curious, is this when you're running `test` or `test_watch`?

**Comment by on3iro** [comment]
@bcoe It happens when i use `npm run coverage` which invokes `npm run test`.
This happened on my windows machine in the office and on our gitlabCI server as well. The weird thing is that I checked out the repository at home on my xubuntu machine, cleared the include/exclude rules out of my babelrc and suddenly got the expected coverage.

Clearing/changing include/exclude rules did not help on the windows machine though. Whenever I changed something the coverage output was different and in most cases even more restricted than before. Even more weird: after a `git reset --hard HEAD` the coverage output was different **again** and did not match the ouput I had before changing any exclude/include rules.
